File: 66. Plus One/plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry = 1
        for i in range(len(digits)-1, -1, -1):
            digits[i] += carry
            carry = digits[i] / 10
            digits[i] %= 10
        if carry == 1:
            digits.insert(0,1)
        return digits

File: 66. Plus One/plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry=1;
        for(vector<int>::reverse_iterator it=digits.rbegin(); it!=digits.rend(); it++){
            *it += carry;
            carry = *it/10;
            *it %= 10;
        }
        if(carry)
            digits.insert(digits.begin(),1);
        return digits;
    }
};
